File: chap2/Temperature.c
// Ces programmes sont sous licence CeCILL-B V1.
#include <stdio.h>

int main() 
{
   int jour;
   double somme, temperature;
   somme = 0;
   for (jour = 1; jour <= 7; jour = jour + 1) 
   {
      scanf("%lf", &temperature);
      somme = somme + temperature;
   }
   printf ("%lf\n",somme / 7.0);
}

File: chap2/Temperature.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let _ =
  let nbJours = 7 in
  let somme = ref 0. in (* "ref" indique une variable modifiable *)
  for loop = 1 to nbJours do
   let temperature = float_of_string (input_line stdin) in
   somme := !somme +. temperature
  done;
  print_float (!somme /. float nbJours);
  print_newline ()

